Hudl.Mjolnir/Config/MjolnirConfiguration.cs

```diff
@@ -123,6 +125,14 @@ public BreakerConfiguration GetBreakerConfiguration(string key = null)
             return BreakerConfigurations.TryGetValue(key, out breakerConfiguration) ?
                 breakerConfiguration : DefaultBreakerConfiguration;
         }
-        
+
+
+        /// <summary>
+        /// Allows subscribtions for configuration change. Whenever any property change in MjolnirConfig all 
+        /// subscribers are being notified of that change.
+        /// </summary>
+        /// <param name="observer">Mjolnir configuration observer which is acting upon configuration change</param>
+        /// <returns></returns>
+        public abstract IDisposable Subscribe(IObserver<MjolnirConfiguration> observer);
```

Comment:
Is it possible to have this class (`MjolnirConfiguration`) handle all of the observer logic and make this `internal` instead of `public abstract`? It looks like most of the implementations of this are the same - do inheriting classes need their own control over how the subscriptions work?